Replace (g) and (r) with icons

JoachimTislov opened the pull request:
Closes: #1288 

Before: 

![image](https://github.com/user-attachments/assets/811bcc32-9322-4840-9d66-4e2e8e225e1a)

After:

![image](https://github.com/user-attachments/assets/57b5ce24-3814-450b-beab-6f4166a0a3cd)

meling commented on public/src/components/ComponentsHelpers.ts:
```diff
@@ -74,19 +73,24 @@ export const generateRow = (
     for (const assignment of assignments) {
-        if (group && assignment.isGroupLab) {
-            base.push({ value: `${assignment.name} (g)`, onClick: () => actions.review.setAssignmentID(assignment.ID) })
+        const cell: CellElement = { iconClassName: "fa fa-users", value: assignment.name, onClick: () => actions.review.setAssignmentID(assignment.ID) }
```
This logic should be simplified. First decide the value of `iconClassName`. Then construct the `cell` element and push to base only once. We shouldn’t need to use continue here.

JoachimTislov replied:
I understand, but it was quite difficult when i first attempted to simplify the logic. I can try again when I have food in my stomach.

JoachimTislov replied:
And be aware, there so much logic on the frontend which should be optimized.. or refactored

JoachimTislov replied:
For example this one, in the same file:

```typescript
if (isGroup && isGroupLab) {
    // If we're dealing with a group assignment and a group, we should try to find a group submission
    submission = submissions.ForGroup(enrollment)?.find(s => s.AssignmentID.toString() === assignmentID)
}

if (isEnrollment) {
    if (isGroupLab && enrollment.groupID === 0n) {
        // If we're dealing with a group assignment, and the enrollment is not part of a group
        // we should try to find an individual submission instead
        submission = submissions.ForUser(enrollment)?.find(s => s.AssignmentID.toString() === assignmentID)
    } else if (isGroupLab && !individual) {
        // If we're dealing with a group assignment, and the user is not viewing individual submissions
        submission = submissions.ForGroup(enrollment)?.find(s => s.AssignmentID.toString() === assignmentID)
    } else {
        submission = submissions.ForUser(enrollment)?.find(s => s.AssignmentID.toString() === assignmentID)
    }
}
```

meling reviewed (changes requested):
I like this redesign. @JosteinLindhom any concerns?

JosteinLindhom commented:
I like it as well. 

Not sure it's necessary, but we could add a title to the icons such that hovering it shows what it means:

```html
<i class="fa fa-unlock" title="Released">
<i class="fa fa-lock" title="Not Released">
```

JoachimTislov commented:
Replacing all the N/A with a minus improves the visual of the table.

Latest commit;

![image](https://github.com/user-attachments/assets/81055f89-6070-49b1-bb88-98430ca9981c)

JosteinLindhom commented:
Maybe tone down the darkness of the minus a little, other than that it's very nice!

meling commented on public/src/components/ComponentsHelpers.ts:
```diff
@@ -69,24 +69,25 @@ export const generateRow = (
-        }
-        if (!group) {
-            base.push({ value: assignment.isGroupLab ? `${assignment.name} (g)` : assignment.name, onClick: () => actions.review.setAssignmentID(assignment.ID) })
+        const cell: CellElement = { value: assignment.name, onClick: () => actions.review.setAssignmentID(assignment.ID) }
```
I find the logic below a bit difficult to reason about. I think the following is better. It should be the first statement in the for loop:
```typescript
// If we are viewing by group, ignore all non-group assignments
if (viewByGroup && !assignment.isGroupLab) {
    continue
}
```
Then you can replace the two if statements below with:
```typescript
if (assignment.isGroupLab) {
    cell.iconTitle = "Group"; cell.iconClassName = "fa fa-users"
}

meling reviewed (changes requested):
A small logic simplification please.

JoachimTislov commented:
I replaced the fat minus icon with the normal minus: "-" on the keyboard. Also added a user icon for invdividual labs, not sure if this is redundant, as you would asume its individual if its not a group assignment/lab.

![image](https://github.com/user-attachments/assets/e3cf95a1-079d-4d43-89e1-6612f24ba174)

meling commented on public/src/__tests__/ComponentsHelpers.test.ts:
```diff
@@ -14,10 +14,10 @@ describe("generateRow", () => {
             // Group submission:
             // - Submission ID: 4, Assignment ID: 4
-            desc: "Enrollment{ID: 1, groupID: 1} should have rows {1, N/A, 3, 4}",
+            desc: "Enrollment{ID: 1, groupID: 1} should have rows {1, -, 3, 4}",
```
What about making a global variable `NotAvailableSymbol` or `MissingSymbol` somewhere so that we can easily change the symbol?

meling replied:
Here are some ideas from ChatGPT:

<head></head><p class="p1" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;">Ah, got it — for a table display where you want a visual, possibly<span class="Apple-converted-space"> </span><i>aesthetic</i><span class="Apple-converted-space"> </span>way to show missing/unavailable data, emojis can work well. Here are several emoji options that might fit different tones (neutral, playful, serious, etc.):</p><p class="p2" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="s1"></span></p><hr style="font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><p style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"></p><p class="p3" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><b>Neutral / Minimal</b><b></b></p><p class="p4" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="Apple-tab-span"></span>•<span class="Apple-tab-span"><span class="Apple-converted-space"> </span></span><span class="s2">⬜</span><span class="Apple-converted-space"> </span>White Large Square — simple placeholder</p><p class="p4" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="Apple-tab-span"></span>•<span class="Apple-tab-span"><span class="Apple-converted-space"> </span></span><span class="s2">▫️</span><span class="Apple-converted-space"> </span>White Small Square — smaller, subtle</p><p class="p4" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="Apple-tab-span"></span>•<span class="Apple-tab-span"><span class="Apple-converted-space"> </span></span><span class="s2">➖</span><span class="Apple-converted-space"> </span>Minus — minimal and balanced</p><p class="p4" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="Apple-tab-span"></span>•<span class="Apple-tab-span"><span class="Apple-converted-space"> </span></span><span class="s2">…</span><span class="Apple-converted-space"> </span>Horizontal Ellipsis (technically punctuation, but often used like an emoji)</p><p class="p2" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="s1"></span></p><hr style="font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><p style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"></p><p class="p3" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><b>Conceptual / Abstract</b><b></b></p><p class="p4" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="Apple-tab-span"></span>•<span class="Apple-tab-span"><span class="Apple-converted-space"> </span></span><span class="s2">❔</span><span class="Apple-converted-space"> </span>White Question Mark — data is unknown</p><p class="p4" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="Apple-tab-span"></span>•<span class="Apple-tab-span"><span class="Apple-converted-space"> </span></span><span class="s2">❓</span><span class="Apple-converted-space"> </span>Red Question Mark — more prominent</p><p class="p4" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="Apple-tab-span"></span>•<span class="Apple-tab-span"><span class="Apple-converted-space"> </span></span><span class="s2">🚫</span><span class="Apple-converted-space"> </span>Prohibited — intentionally unavailable or restricted</p><p class="p4" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="Apple-tab-span"></span>•<span class="Apple-tab-span"><span class="Apple-converted-space"> </span></span><span class="s2">⚪</span><span class="Apple-converted-space"> </span>White Circle — empty, potential to be filled</p><p class="p4" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="Apple-tab-span"></span>•<span class="Apple-tab-span"><span class="Apple-converted-space"> </span></span><span class="s2">🕳️</span><span class="Apple-converted-space"> </span>Hole — suggests missing or void</p><p class="p2" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="s1"></span></p><hr style="font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><p style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"></p><p class="p3" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><b>Data-Oriented / Symbolic</b><b></b></p><p class="p4" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="Apple-tab-span"></span>•<span class="Apple-tab-span"><span class="Apple-converted-space"> </span></span><span class="s2">📭</span><span class="Apple-converted-space"> </span>Empty Mailbox — “no data received”</p><p class="p4" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="Apple-tab-span"></span>•<span class="Apple-tab-span"><span class="Apple-converted-space"> </span></span><span class="s2">📄</span><span class="Apple-converted-space"> </span>with a strikethrough (not an emoji, but visually,<span class="Apple-converted-space"> </span><span class="s2">📄🚫</span><span class="Apple-converted-space"> </span>could work)</p><p class="p4" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="Apple-tab-span"></span>•<span class="Apple-tab-span"><span class="Apple-converted-space"> </span></span><span class="s2">🗳️</span><span class="Apple-converted-space"> </span>Ballot Box — “to be filled in”</p><p class="p4" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="Apple-tab-span"></span>•<span class="Apple-tab-span"><span class="Apple-converted-space"> </span></span><span class="s2">🔲</span><span class="Apple-converted-space"> </span>Black Square Button — clearly “not filled”</p><p class="p4" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="Apple-tab-span"></span>•<span class="Apple-tab-span"><span class="Apple-converted-space"> </span></span><span class="s2">📉</span><span class="Apple-converted-space"> </span>Data Missing (figuratively)</p><p class="p2" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="s1"></span></p><hr style="font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><p style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"></p><p class="p3" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><b>Playful / Friendly</b><b></b></p><p class="p4" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="Apple-tab-span"></span>•<span class="Apple-tab-span"><span class="Apple-converted-space"> </span></span><span class="s2">🕸️</span><span class="Apple-converted-space"> </span>Cobweb — “nobody’s touched this in a while”</p><p class="p4" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="Apple-tab-span"></span>•<span class="Apple-tab-span"><span class="Apple-converted-space"> </span></span><span class="s2">👻</span><span class="Apple-converted-space"> </span>Ghost — data ghosted us</p><p class="p4" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="Apple-tab-span"></span>•<span class="Apple-tab-span"><span class="Apple-converted-space"> </span></span><span class="s2">🫥</span><span class="Apple-converted-space"> </span>Dotted Face — fading out, something’s not there</p><p class="p4" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="Apple-tab-span"></span>•<span class="Apple-tab-span"><span class="Apple-converted-space"> </span></span><span class="s2">😶</span><span class="Apple-converted-space"> </span>Face Without Mouth — no comment / nothing provided</p><p class="p2" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="s1"></span></p><hr style="font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><p style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"></p><p class="p3" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><b>My Recommendations by Tone</b></p>
Tone | Emoji | Comment
-- | -- | --
Neutral | ➖, ⬜, ▫️ | Minimal disruption to table layout
Informative | ❔, ❓, … | Immediately signals unknown
Restrictive | 🚫, 🕳️ | Good if values are intentionally unavailable
Playful | 👻, 😶, 🫥 | If tone permits a bit of personality

<p class="p1" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><br></p><p class="p2" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><span class="s1"></span></p><hr style="font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"><p style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;"></p><p class="p3" style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration: none;">Let me know if your table is monochrome, compact, or being rendered in a specific UI toolkit (e.g., terminal, web, GUI framework), and I can refine the suggestions!</p>

meling replied:
I kind of like the more discrete symbols like the dash, but perhaps not as black as the first one. Maybe a grey one would be lighter on the eyes. Or you could use a playful one... like the empty hole.

JoachimTislov replied:
Im not sure if font awesome has these icons, but I can have a look

meling reviewed (changes requested):
Looks good to me. One minor tweak that I think we should add though: instead of having many uses of a simple symbol like this `-`, we should introduce a global variable where we can change it for all instances. See comment inline.

@Glattshaved Any thoughts on the visuals?

JoachimTislov commented:
Latest commit;

![image](https://github.com/user-attachments/assets/38aa6f83-fa87-43d4-a957-446c064403d4)

meling commented on public/src/components/Icons.tsx:
```diff
@@ -0,0 +1,6 @@
+import { Icon } from '../Helpers'
+
+export const Icons = {
+    GreyDash: <i className={Icon.DASH} />,
```
It is better to name the "style" something like `NotAvailable`/`NoSubmission` instead of `GreyDash` to give it semantic meaning in the code. Then if we later decide to change it to another icon of some sort, we only need to change it here and not everywhere we use the not available or no submission.

meling reviewed (approved):
Please see suggested fix.
